Traceback (most recent call last):
  File "/home/richard/.virtualenvs/etcdocs/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/richard/.virtualenvs/etcdocs/lib/python3.8/site-packages/nbclient/client.py", line 1087, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/richard/.virtualenvs/etcdocs/lib/python3.8/site-packages/nbclient/util.py", line 74, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/richard/.virtualenvs/etcdocs/lib/python3.8/site-packages/nbclient/util.py", line 53, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/richard/.virtualenvs/etcdocs/lib/python3.8/site-packages/nbclient/client.py", line 540, in async_execute
    await self.async_execute_cell(
  File "/home/richard/.virtualenvs/etcdocs/lib/python3.8/site-packages/nbclient/client.py", line 832, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/home/richard/.virtualenvs/etcdocs/lib/python3.8/site-packages/nbclient/client.py", line 740, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply['content'])
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Draw heatmap of change

df = pd.DataFrame()

x_axis = []
y_axis = []

for i in range(1, 16):
    for j in range(1, 16):
        x = (j-1)*10000 + 1
        y = (i-1)*150000 
        x_axis.append(x)
        y_axis.append(y)

# For some reason this is inverted
df = pd.DataFrame(data={'Object Record (O1 to O2100000)': y_axis, 'Object Record (10,000 per square) ': x_axis, 'Data': change_identifier_bins})
objects = df.pivot("X", "Y", "Data")

sns.set_style("dark")
sns.color_palette("deep")

# Draw a heatmap with the numeric values in each cell
f, ax = plt.subplots(figsize=(10, 7))

heatmap = sns.heatmap(objects, annot=True, fmt="d", linewidths=.5, ax=ax, cmap="vlag")

------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mKeyError[0m                                  Traceback (most recent call last)
[0;32m~/.virtualenvs/etcdocs/lib/python3.8/site-packages/pandas/core/indexes/base.py[0m in [0;36mget_loc[0;34m(self, key, method, tolerance)[0m
[1;32m   2894[0m             [0;32mtry[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m-> 2895[0;31m                 [0;32mreturn[0m [0mself[0m[0;34m.[0m[0m_engine[0m[0;34m.[0m[0mget_loc[0m[0;34m([0m[0mcasted_key[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   2896[0m             [0;32mexcept[0m [0mKeyError[0m [0;32mas[0m [0merr[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;32mpandas/_libs/index.pyx[0m in [0;36mpandas._libs.index.IndexEngine.get_loc[0;34m()[0m

[0;32mpandas/_libs/index.pyx[0m in [0;36mpandas._libs.index.IndexEngine.get_loc[0;34m()[0m

[0;32mpandas/_libs/hashtable_class_helper.pxi[0m in [0;36mpandas._libs.hashtable.PyObjectHashTable.get_item[0;34m()[0m

[0;32mpandas/_libs/hashtable_class_helper.pxi[0m in [0;36mpandas._libs.hashtable.PyObjectHashTable.get_item[0;34m()[0m

[0;31mKeyError[0m: 'X'

The above exception was the direct cause of the following exception:

[0;31mKeyError[0m                                  Traceback (most recent call last)
[0;32m<ipython-input-10-958635966c7b>[0m in [0;36m<module>[0;34m[0m
[1;32m     15[0m [0;31m# For some reason this is inverted[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[1;32m     16[0m [0mdf[0m [0;34m=[0m [0mpd[0m[0;34m.[0m[0mDataFrame[0m[0;34m([0m[0mdata[0m[0;34m=[0m[0;34m{[0m[0;34m'Object Record (O1 to O2100000)'[0m[0;34m:[0m [0my_axis[0m[0;34m,[0m [0;34m'Object Record (10,000 per square) '[0m[0;34m:[0m [0mx_axis[0m[0;34m,[0m [0;34m'Data'[0m[0;34m:[0m [0mchange_identifier_bins[0m[0;34m}[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 17[0;31m [0mobjects[0m [0;34m=[0m [0mdf[0m[0;34m.[0m[0mpivot[0m[0;34m([0m[0;34m"X"[0m[0;34m,[0m [0;34m"Y"[0m[0;34m,[0m [0;34m"Data"[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     18[0m [0;34m[0m[0m
[1;32m     19[0m [0msns[0m[0;34m.[0m[0mset_style[0m[0;34m([0m[0;34m"dark"[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/.virtualenvs/etcdocs/lib/python3.8/site-packages/pandas/core/frame.py[0m in [0;36mpivot[0;34m(self, index, columns, values)[0m
[1;32m   6676[0m         [0;32mfrom[0m [0mpandas[0m[0;34m.[0m[0mcore[0m[0;34m.[0m[0mreshape[0m[0;34m.[0m[0mpivot[0m [0;32mimport[0m [0mpivot[0m[0;34m[0m[0;34m[0m[0m
[1;32m   6677[0m [0;34m[0m[0m
[0;32m-> 6678[0;31m         [0;32mreturn[0m [0mpivot[0m[0;34m([0m[0mself[0m[0;34m,[0m [0mindex[0m[0;34m=[0m[0mindex[0m[0;34m,[0m [0mcolumns[0m[0;34m=[0m[0mcolumns[0m[0;34m,[0m [0mvalues[0m[0;34m=[0m[0mvalues[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   6679[0m [0;34m[0m[0m
[1;32m   6680[0m     _shared_docs[

[0;32m~/.virtualenvs/etcdocs/lib/python3.8/site-packages/pandas/core/reshape/pivot.py[0m in [0;36mpivot[0;34m(data, index, columns, values)[0m
[1;32m    461[0m         [0;32melse[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m    462[0m             [0mindex[0m [0;34m=[0m [0mcom[0m[0;34m.[0m[0mconvert_to_list_like[0m[0;34m([0m[0mindex[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 463[0;31m             [0mindex[0m [0;34m=[0m [0;34m[[0m[0mdata[0m[0;34m[[0m[0midx[0m[0;34m][0m [0;32mfor[0m [0midx[0m [0;32min[0m [0mindex[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    464[0m [0;34m[0m[0m
[1;32m    465[0m         [0mdata_columns[0m [0;34m=[0m [0;34m[[0m[0mdata[0m[0;34m[[0m[0mcol[0m[0;34m][0m [0;32mfor[0m [0mcol[0m [0;32min[0m [0mcolumns[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m

[0;32m~/.virtualenvs/etcdocs/lib/python3.8/site-packages/pandas/core/reshape/pivot.py[0m in [0;36m<listcomp>[0;34m(.0)[0m
[1;32m    461[0m         [0;32melse[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m    462[0m             [0mindex[0m [0;34m=[0m [0mcom[0m[0;34m.[0m[0mconvert_to_list_like[0m[0;34m([0m[0mindex[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 463[0;31m             [0mindex[0m [0;34m=[0m [0;34m[[0m[0mdata[0m[0;34m[[0m[0midx[0m[0;34m][0m [0;32mfor[0m [0midx[0m [0;32min[0m [0mindex[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    464[0m [0;34m[0m[0m
[1;32m    465[0m         [0mdata_columns[0m [0;34m=[0m [0;34m[[0m[0mdata[0m[0;34m[[0m[0mcol[0m[0;34m][0m [0;32mfor[0m [0mcol[0m [0;32min[0m [0mcolumns[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m

[0;32m~/.virtualenvs/etcdocs/lib/python3.8/site-packages/pandas/core/frame.py[0m in [0;36m__getitem__[0;34m(self, key)[0m
[1;32m   2904[0m             [0;32mif[0m [0mself[0m[0;34m.[0m[0mcolumns[0m[0;34m.[0m[0mnlevels[0m [0;34m>[0m [0;36m1[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m   2905[0m                 [0;32mreturn[0m [0mself[0m[0;34m.[0m[0m_getitem_multilevel[0m[0;34m([0m[0mkey[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m-> 2906[0;31m             [0mindexer[0m [0;34m=[0m [0mself[0m[0;34m.[0m[0mcolumns[0m[0;34m.[0m[0mget_loc[0m[0;34m([0m[0mkey[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   2907[0m             [0;32mif[0m [0mis_integer[0m[0;34m([0m[0mindexer[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m   2908[0m                 [0mindexer[0m [0;34m=[0m [0;34m[[0m[0mindexer[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m

[0;32m~/.virtualenvs/etcdocs/lib/python3.8/site-packages/pandas/core/indexes/base.py[0m in [0;36mget_loc[0;34m(self, key, method, tolerance)[0m
[1;32m   2895[0m                 [0;32mreturn[0m [0mself[0m[0;34m.[0m[0m_engine[0m[0;34m.[0m[0mget_loc[0m[0;34m([0m[0mcasted_key[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m   2896[0m             [0;32mexcept[0m [0mKeyError[0m [0;32mas[0m [0merr[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m-> 2897[0;31m                 [0;32mraise[0m [0mKeyError[0m[0;34m([0m[0mkey[0m[0;34m)[0m [0;32mfrom[0m [0merr[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m   2898[0m [0;34m[0m[0m
[1;32m   2899[0m         [0;32mif[0m [0mtolerance[0m [0;32mis[0m [0;32mnot[0m [0;32mNone[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;31mKeyError[0m: 'X'
KeyError: 'X'

